src/arith_uint256.h

```diff
@@ -26,6 +26,7 @@ class base_uint
 protected:
     static_assert(BITS / 32 > 0 && BITS % 32 == 0, "Template parameter BITS must be a positive multiple of 32.");
     static constexpr int WIDTH = BITS / 32;
+    /** Fully little-endian byte ordering on little-endian architectures. */
```

Comment:
> > Maybe just say number is stored in the array with least significant bits first and most significant bits last.
> 
> That is incorrect. The least significant _byte_ is stored first for `uint32_t` on the little-endian archs. If one wants to avoid mentioning architecture, one could possibly say "the number is stored in an array with the least significant uint32_t chunk first".

It is correct, it's just saying the highest bits of the number are in the last array element and the lowest bits of the number are in the first array element. But you could definitely say chunks instead of bits if you prefer that.

> > The current comment describing byte ordering on one specific architecture seems like an indirect way of describing the array when array elements aren't bytes, and the class itself is architecture independent.
> 
> Do you mean mentioning architecture on a protected member instills fear that the users of the class will be exposed to architecture dependent behavior?

It's just confusing and indirect.

Reply:
> It is correct, it's just saying the highest bits of the number are in the last array element and the lowest bits of the number are in the first array element. But you could definitely say chunks instead of bits if you prefer that.

It is correct if by "bits" you mean 32-bit chunks together I guess, then the lowest is at the beginning and the highest at the end. I was somehow wrongly interpreting you as meaning *the* highest bit (singular) at the end + *the* lowest bit at the beginning.

> It's just confusing and indirect.

Would this feel less confusing?
```C++
    /** Number is stored as fully little-endian on little-endian archs. */
    uint32_t pn[WIDTH];
```
..? Then it at least specifies what is being stored (but it's the only data-member of `base_uint` so could be considered overly verbose IMO). Kept it short to stay within 80 char line.